settings.php

```diff
@@ -142,6 +142,13 @@
             $CFG->wwwroot . '/auth/saml2/regenerate.php'
             ));
 
+    $settings->add(new admin_setting_configtext(
+        'auth_saml2/privatekeypass',
+        get_string('privatekeypass', 'auth_saml2'),
+        get_string('privatekeypass_help', 'auth_saml2'),
+        get_site_identifier(),
```

Comment:
I'm wondering wether it would be better if this defaulted to an empty string, and if it was empty then it dynamically defaults back to get_site_identifier()?

I'm thinking around when data gets refreshed to other environments at the moment this should correctly break, but with this patch it will now work. Maybe this is a feature, maybe this is a bug. From a security perspective I'd prefer a refresh script to just force set the value of the privatekeypass and not inherit it and leak a private key

Reply:
If the privatekeypass is supposed to be "secret" I'm not convinced we should be using the site identifier here at all? - Most uses of the site_identifer (backup xml etc) use an MD5 of the siteidentifier, but that's not guaranteed I think. The existing use of site identifier is why I implemented this as a setting in the end - rather than just a hard-coded value set at the db level.

If it should be secret we could create a hash stored in the config table on install (with no UI) that is only used by the saml private key instead of relying on the site identifier. (existing installs would just copy the existing site identifier into this new config var.)

Also - when data gets "refreshed" from a prod to testing site the siteidentifier is usually pulled across from the prod database at the same time, (unless using some form of plugin for transferring data instead of a db level copy.) the Moodle site doesn't regenerate the siteidentifier unless you trigger a hub registration and it finds a url and siteidentifier don't match.

that sound any better?

Reply:
I just double checked and I was wrong, the site id isn't reset from a 'raw' refresh, this is done in the local_datacleaner plugin. For some reason I thought if the domain changed the siteid got reset.

I don't think this config item is any more sensitive than some other critical admin settings, like oauth tokens etc so I don't think it's worth implementing a special admin setting for this, but I do think swapping it to use admin_setting_configpasswordunmask is enough and that's the main thing I'd like to see to land. That opens the door for some tighter security across the board for those that need it via MDL-66445 which we'll tackle shortly

Reply:
cool - will be nice to see MDL-66445 land at some point. I've fixed those and rebased for you to take another look.